### ISSUE TITLE
Need for 2nd maintainer

### ISSUE THREAD

**Issue by vlcinsky** [jazzband/tablib#329]
As @hugovk mentioned in PR #310, we have a maintainer issue with tablib (similarly with `records`).

Facts:
- @kennethreitz is Development Lead (592 commits)
- @iurisilvio is Core Contributor (14 commits)
- There are 20 authors mentioned in AUTHORS.
  - @lbeltrame and @durden 13 commits
  - @f4nt, @claudep and @msabramo 5 commits or more
  - 10 other with 3 commits
  - 6 others with 2 commits
  - 47 other with 1 commit
- Issues: 107 closed, 68 open
  - last 25 open issues are younger than 12 months
- PR 135 closed, 18 open
  - 16 open PRs are younger than 12 months
  - good number of those PRs seem to fix open issues
  - last PR accepted is around 7 month old
- downloads/pip installs
  - 48 thousands in last 30 days (see table in PR #310)
- last version is `0.12.1` released on Sep 1, 2017

To summarize:
- There is clear interest on `tablib` from users as well as from contributors.
- Maintenance got somehow stuck since September last year.

Proposal for next actions:
- @kennethreitz , @iurisilvio are you willing to accept another maintainer or even hand over to someone?
- others: is there anyone feeling like becoming maintainer?

**Comment by claudep** [comment]
Other option: transfer project to https://jazzband.co/

**Comment by claudep** [comment]
I've now applied some patches to my fork: https://github.com/claudep/tablib
"Unfriendly" forks should be a last resort solution, so I really hope we'll find another solution in the longer term, but I intend to temporarily maintain my fork until this gets resolved.

**Comment by sergei-maertens** [comment]
@claudep are releasing your fork on PyPI as well, or only keeping a github fork around?

**Comment by claudep** [comment]
I'm still reluctant to go forward with the fork, as I find it so weird that an eminent developer like Kenneth is not even able to say a word about this package maintenance. If someone reading this knows him personally, please take the time to talk to him!

**Comment by sergei-maertens** [comment]
I've reached out to Kenneth through a Twitter DM, that's my best bet to get in touch.

Other than that, I'd be willing to step up to maintain this, I think it must be possible to set some time aside on a Friday afternoon at $dayjob to get things sorted out.

**Comment by sergei-maertens** [comment]
Kenneth has replied through Twitter - so we're making progress here. I'm totally okay with taking up maintenance, but also don't want to be 'snapping' away this, so if anyone else wants to step up, maybe it wouldn't be bad to have multiple people added with write permissions on github and PyPI.

**Comment by claudep** [comment]
I think everyone will welcome any responsive maintainer, so feel free to continue what you started.

**Comment by vlcinsky** [comment]
@sergei-maertens go for it. Thanks for contacting Kenneth.

How about the other projects (records, etc.). They are quite interdependent and some long-term issues were blocking progress in other projects.

To me it seems like a fleet of ships, which shall be somehow coordinated and would benefit with spare commander when the captain is busy with other stuff.

**Comment by sergei-maertens** [comment]
I'm not familiar with the other projects (well, except for requests, as a user), so I'd limit myself only to tablib. I'm fine with keeping other projects in mind when working on tablib - as long as I can be properly informed, but I'm sure that will work out.

As for the other projects - I had success reaching out via Twitter, but I'm also hesitant to say 'just DM him there' for obvious reasons. Maybe on my next contact I should just ask to review the call-for-maintainers issues on the other projects? Is there any other project than records that needs more maintainers?

**Comment by sergei-maertens** [comment]
Unfortunately not

**Comment by sergei-maertens** [comment]
It is not - I haven't heard back from Kenneth

**Comment by claudep** [comment]
Sergei, would you suggest a way forward? Should we create some sort of official unofficial common fork  :grimacing:

**Comment by claudep** [comment]
Oh, @iurisilvio, do you plan to be (regularly) active again? What's your take about this issue?

**Comment by sergei-maertens** [comment]
Interested to hear what @iurisilvio has to say about it, otherwise I would indeed go forward with an 'official' unofficial fork and hope that the changes can then be merged back upstream at some point.

The main issue right now is that we can't release new versions on PyPI. @iurisilvio - can you add maintainers on PyPI, or is that something only Kenneth can do?

**Comment by kennethreitz** [comment]
I’m not opposed to it

**Comment by benzkji** [comment]
I'd strongly vote for jazzband. curious how this continues....

**Comment by antoine-gallix** [comment]
@kenneth-reitz , now that you have put efforts in the past to make that library, that you have users using it, contributors willing to contribute and people willing to maintain it, and that you understandably have no time nor interest to take care of it anymore, acknowledging the latter fact and making a move to pass on the responsibility would be a very nice move to let the project continue it's life. People here on in an organization like jazzband are willing to do that work. Independent forking of the project could move the project forward, but would have to regain the user base, which is quite close to letting the project die like it is now the case. Both case would be quite a shame, and would not make honor to all the work done until now. I really liked the API and was using it at work, but because of a small bug that was just waiting for a maintainer to merge the fix for months, we had to reinvent the wheel and recreate a subset of the API, and then moved to using pandas, which is absurd for the use that we have of it. I'm relatively new to open source, and I can't understand how the creator of a project can possibly prefer it's project to die than letting it live it's life in the hands of other people.

**Comment by timofurrer** [comment]
Thanks to all of you for reaching out. It seems like we all want the same thing. 
As you may know Kenneth changed his account to an organization, which as of today, many people are part of, who are willing to help out maintaining repos which didn't get a lot of attention lately - including tablib.

We'll try to get things running again here. If anyone wants to help out as a maintainer / contributor for tablib, please reach out to us at: https://github.com/kennethreitz/team/issues/17

Or have a chat at our discord: https://discord.gg/sps3Fj

**Comment by claudep** [comment]
I guess the ticket could be closed now, thanks @timofurrer for taking the lead! Don't hesitate to reach to me if you need help.

**Comment by isedwards** [comment]
Could someone update this issue on how the ownership moved to @vinayak-mehta (just so that we understand what happened)?

Was there not any support for moving the project to Jazzband?
- https://github.com/jazzband
- https://jazzband.co/

**Comment by hugovk** [comment]
Kenneth decided to find a new home for all his repos, and Vinayak kindly volunteered for tablib. 

https://github.com/not-kennethreitz/team/issues/21

https://github.com/not-kennethreitz/team/issues/21#issuecomment-512533194

+1 to Jazzband, it'd would be a good home, a lead developer (or developers) are still needed, and Jazzband can help provide some support for everyday admin/maintenance/release/infra stuff.

This has also been proposed for other repos in https://github.com/not-kennethreitz/team/issues/21.

**Comment by claudep** [comment]
Transfering to Vinayak did not improve the situation at all.
@iurisilvio, @timofurrer, @vinayak-mehta, you were all kind trying to maintain tablib, now could you help moving the project to Jazzband, please?

**Comment by vinayak-mehta** [comment]
Sorry I haven't been able to spend time here. I can transfer ownership to Jazzband. Or should we perhaps move it to PSF where some of the other projects went?

**Comment by vinayak-mehta** [comment]
I see only requests went to PSF. Jazzband might be a good home, albeit having mostly Django-specific projects. What do @iurisilvio and @timofurrer think?

**Comment by vinayak-mehta** [comment]
I would also like to suggest [camelot-dev](https://github.com/camelot-dev) as a possible alternative. I volunteered to maintain tablib was because of its alignment with camelot. We can possibly change the org name to something else and build a home for projects that deal with tabular data (maybe other types of data too). We can add more maintainers to the org, right now there's only @dimitern and me.

**Comment by hugovk** [comment]
The benefit of Jazzband is there's more backup in case people are too busy. They can especially help with things like releases (and [partially automating releases](https://jazzband.co/about/releases)).

**Comment by timofurrer** [comment]
> I see only requests went to PSF. Jazzband might be a good home, albeit having mostly Django-specific projects. What do @iurisilvio and @timofurrer think?

I'm no longer maintaining here nor do I feel responsible. 
However, I totally agree that this repository should be transferred to Jazzband - like a lot others should have been during the transfer of Kenneth's repos ...

**Comment by vinayak-mehta** [comment]
Thanks for your thoughts! I've initiated a transfer of tablib to Jazzband :)

**Comment by vinayak-mehta** [comment]
Not quite yet, bear with me :sweat_smile: Should we create a tablib team at Jazzband who will have write access to master? (considering tablib is used by a lot of other projects) Right now, every member will be able to push to master. 

I think we'll have to ask Kenneth to add `jazzband` as a PyPI maintainer as mentioned here: https://jazzband.co/about/releases

**Comment by vinayak-mehta** [comment]
@claudep @hugovk You could part of that team.

**Comment by claudep** [comment]
As for me, I'm trusting Jazzband members. If anyone tries to abuse the system by not respecting the code of conduct, he'll probably be excluded and any commit can be reverted anytime.

**Comment by vinayak-mehta** [comment]
Cool, I guess that makes sense since we haven't heard cases of people not respecting the Jazzband code of conduct.

**Comment by chfw** [comment]
Simply protect master, should do.

**Comment by vinayak-mehta** [comment]
I did the transfer by not selecting any of the boxes just to check if that'll work and it did :| Can everyone still write to master? I guess those check boxes were for repository settings, because I can't see them now.

**Comment by claudep** [comment]
@timofurrer, we need you to add jazzband account to PyPI project as maintainer role.

**Comment by timofurrer** [comment]
done 👍 🎉

**Comment by jezdez** [comment]
I think this can be closed now that the project has been moved to Jazzband. @claudep and @hugovk are the current project leads. See #378 for more details. Welcome!
